src/generator.cs

```diff
@@ -5395,6 +5407,17 @@ public void PrintPreserveAttribute (ICustomAttributeProvider mi)
 			print ("[Preserve]");
 	}
 
+	// Function to check if PrintAdviceAttribute is present and
+	// generate/print the same attribute as in bindings
+	public void PrintAdviceAttribute (ICustomAttributeProvider mi)
+	{
+		var p = AttributeManager.GetCustomAttribute<AdviceAttribute> (mi);
+		if (p == null)
+			return;
+
+		print ($"[Advice (\"{p.Message}\")]");
```

Comment:
You should escape any quotes from `p.Message`:

    print ($"[Advice (@\"{p.Message.Replace ("\"", "\"\"")}\")]");